src/sentry/search/events/datasets/base.py

```diff
@@ -15,6 +18,10 @@ class DatasetConfig(abc.ABC):
     non_nullable_keys: set[str] = set()
     missing_function_error: ClassVar[type[Exception]] = InvalidSearchQuery
 
+    @abc.abstractmethod
+    def __init__(self, builder: "BaseQueryBuilder"):
```

Comment:
You've imported it above:

```suggestion
    def __init__(self, builder: BaseQueryBuilder):
```

Reply:
That doesn't work, its only imported for type checking

Reply:
add `from __future__ import annotations` to the file